### ISSUE TITLE
JQuery 1.2 < 3.5.0 XSS

### ISSUE THREAD

**Issue by takenek** [owncloud/core#37463]
Hello,
Please update jquery in OwnCloud:
./core/vendor/jquery/dist/jquery.min.js

Description
According to the self-reported version in the script, the version of JQuery hosted on the remote web server is greater than or equal to 1.2 and prior to 3.5.0. It is, therefore, affected by a cross site scripting vulnerability.
Solution
Upgrade to JQuery version 3.5.0 or later.
See Also
https://nvd.nist.gov/vuln/detail/CVE-2020-11022

I check and in OwnCloud 10.4.1 there is old version in this directory: /core/vendor/jquery/dist/jquery.min.js.

Best Regards
TaKeN

**Comment by takenek** [comment]
Same problem in newlatest beta version:
https://download.owncloud.org/community/testing/owncloud-10.5.0beta2.tar.bz2

**Comment by ho4ho** [comment]
@takenek https://raw.githubusercontent.com/owncloud/core/master/.github/issue_template.md

> For reporting potential security issues please see https://owncloud.org/security/

Better ask by above?

**Comment by mmattel** [comment]
@micbar fyi

**Comment by ho4ho** [comment]
> Yes, that is the same issue.
> https://github.com/owncloud/core/pull/37596#issuecomment-650571933

@micbar @mmattel Close?

**Comment by mmattel** [comment]
Form my pov, #37596 improves the situation but it is not updating jquery to the newest release...

**Comment by ho4ho** [comment]
But XSS fixed by workaround?

> To workaround the issue without upgrading, adding the following to your code:
> https://github.com/advisories/GHSA-gxr4-xjj5-5px2

**Comment by mmattel** [comment]
@ho4ho 

yes, the xss issue is fixed by the workaround.
As the name "workaround" says, ist is not a clean fix.
Either:
- we close this issue because of the workaround and forget about any jquery upgrade
- we keep it open and use it to upgrade to 3.5.1 (https://code.jquery.com)
- or we close this one and create a new issue (refrencing to this issue) which is imho the cleanest path to go. 

Including that between our really old jquery version and the current release of 3.5.1, a lot of additional fixes have been made...

@micbar, this is something you (or PM) have to decide

**Comment by ho4ho** [comment]
https://github.com/owncloud/core/issues/36237

> JQuery 2.1.4

**Comment by mmattel** [comment]
Yes, that is the correct PR for the long term fix :smile:

**Comment by micbar** [comment]
> Yes, that is the correct PR for the long term fix 😄

It is not that easy unfortunately. Just udating jquery will create a lot of issues.
This would be a major change across the whole platform including all the apps.

**Comment by mmattel** [comment]
Q: Do we have this xss issue in apps too?

**Comment by VicDeo** [comment]
Duplicate of https://github.com/owncloud/core/issues/36237

**Comment by takenek** [comment]
This is not duplicate because jquery-min.js is still used in newlatest version of OwnCloud.

I just create jquery with all patches please try this:
https://taken.pl/jquery/jquery.js
https://taken.pl/jquery/jquery.min.js

Please test it if working good i push to repo.

**Comment by VicDeo** [comment]
https://github.com/advisories/GHSA-gxr4-xjj5-5px2 
>Workarounds
To workaround the issue without upgrading, adding the following to your code:
```
jQuery.htmlPrefilter = function( html ) {
	return html;
};
```
>You need to use at least jQuery 1.12/2.2 or newer to be able to apply this workaround.

https://github.com/owncloud/core/blob/master/core/js/js.js#L1699-L1706
```
/**
 * Make htmlPrefilter an identity function in jQuery 2.1.4.
 *
 * @see https://github.com/advisories/GHSA-gxr4-xjj5-5px2
 */
$.fn.htmlPrefilter = function(html) {
	return html;
};
```


Case closed? ;)

**Comment by AdminValida** [comment]
Hi!

Any updates on https://github.com/owncloud/core/pull/38479 ?

Regards

**Comment by C0rby** [comment]
Hey people,

our approach with the JavaScript libraries so far was to manually patch the issues to best maintain compatibility with the ownCloud apps. Since we didn't change the version number, the automatic scanners think it's the plain old jQuery 1.2 version but this is not the case.

Here is a list of all the issues we have patched so far:

jQuery:

    * [CVE-2020-11022](https://github.com/advisories/GHSA-gxr4-xjj5-5px2)/[CVE-2020-11023](https://nvd.nist.gov/vuln/detail/CVE-2020-11023)
    patched in 10.5.0 https://github.com/owncloud/core/pull/37596
    
    * [CVE-2015-9251](https://nvd.nist.gov/vuln/detail/CVE-2015-9251)
    patched in 10.0.9RC3 https://github.com/owncloud/core/commit/63a14b4eb62a09966ae0166f1b2f48a22c061cf2
    https://github.com/owncloud/core/pull/31972
    We are escaping the text before displaying it.

    * [CVE-2019-11358](https://nvd.nist.gov/vuln/detail/CVE-2019-11358)
    Before the patch there were no vulnerable places in `owncloud/core`. To exploit the vulnerability we would need to pass user controlled data to `$.extend(...)` which we don't.
    Patched here: https://github.com/owncloud/core/pull/38841

    * [CVE-2016-7103](https://nvd.nist.gov/vuln/detail/CVE-2016-7103)
    We are not using the vulnerable component but still patched for prevention of future misuse.
    Patched here https://github.com/owncloud/core/pull/39545

jQuery-ui:

    * [CVE-2021-41182](https://github.com/jquery/jquery-ui/security/advisories/GHSA-9gj3-hwp5-pmwc), [CVE-2021-41183](https://github.com/jquery/jquery-ui/security/advisories/GHSA-j7qv-pgf6-hvh4), [CVE-2021-41184](https://github.com/jquery/jquery-ui/security/advisories/GHSA-gpqq-952q-5327)
    patched in https://github.com/owncloud/core/pull/39451

If we missed something please point us to the issue and we'll fix it.

**Comment by ho4ho** [comment]
Look like false positive by automatic scanner.
